File: src/main/java/dev/kemalyi/bookmarker/bookmark/jpa/Bookmark.java
package dev.kemalyi.bookmarker.bookmark.jpa;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "bookmarks")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Bookmark {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(nullable = false)
    private String title;
    @Column(nullable = false)
    private String url;
    @Column(nullable = false)
    private Instant createdAt;
    private Instant updatedAt;
}

File: bookmarker-frontend/src/components/Bookmark/types/bookmark.ts
export interface Bookmark {
    title: string,
    url: string,
    createdAt: string,
    updatedAt?: string
}

export interface BookmarkData {
    id: number,
    totalPages: number,
    totalElements: number,
    currentPage: number,
    bookmarks: [Bookmark]
}